src/tunnel.cc

```diff
@@ -1211,6 +1211,9 @@ tunnelStart(ClientHttpRequest * http)
 #if USE_DELAY_POOLS
     tunnelState->server.setDelayId(DelayId::DelayClient(http));
 #endif
+
+    request->hier.startPeerClock();
```

Comment:
Calling startPeerClock() while the clock is already running is safe, right?

Would it be better to startPeerClock() inside TunnelStateData constructor, so that we do not miss any (current or future) tunneling cases that may not go through tunnelStart()?

Reply:
Yes, it is safe - only first call has an effect. Ok, lets move it inside the constructor.